src/scripting/game_lua_kernel.cpp

```diff
@@ -3761,11 +3761,7 @@ static int intf_append_ai(lua_State *L)
 	}
 	ai::configuration::expand_simplified_aspects(side_num, cfg);
 	if(added_dummy_stage) {
-		for(auto iter = cfg.ordered_begin(); iter != cfg.ordered_end(); iter++) {
-			if(iter->key == "stage" && iter->cfg["name"] == "empty") {
-				iter = cfg.erase(iter);
-			}
-		}
+		cfg.remove_children("stage", [](const config& stage_cfg) { return stage_cfg["name"] == "empty"; });
```

Comment:
I'm not 100% sure of this, but I think this can probably be reduced to removing _all_ stage children.

Consider that `added_dummy_stage` can only be set to true if there were no stages to begin with. Thus, the only possible way there could be a stage other than the dummy one would be if `expand_simplified_aspects` adds a stage.

I _think_ it can only add a stage if one of two conditions are satisfied: 1. an `ai_algorithm` key is present, or 2. there is no `ai_algorithm` key and also no stage tags. The `ai_algorithm` tag is not supposed to be supported in `append_ai` anyway, if I recall correctly (would have to double-check that as well). So I'm _fairly_ sure it won't add stages in this case.

Reply:
🤔 Worth keeping in mind. I would want to handle that separately, though, since it’s not obvious without testing that that would be the case.